homeassistant/components/fastdotcom/__init__.py

```diff
@@ -40,38 +35,49 @@
 )
 
 
-async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+async def async_setup_platform(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the Fast.com component. (deprecated)."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up the Fast.com component."""
-    conf = config[DOMAIN]
     data = hass.data[DOMAIN] = SpeedtestData(hass)
 
-    if not conf[CONF_MANUAL]:
-        async_track_time_interval(hass, data.update, conf[CONF_SCAN_INTERVAL])
+    async_track_time_interval(hass, data.update, timedelta(hours=DEFAULT_INTERVAL))
 
-    def update(service_call: ServiceCall | None = None) -> None:
+    async def update(service_call: ServiceCall | None = None) -> None:
         """Service call to manually update the data."""
-        data.update()
+        await data.update()
 
     hass.services.async_register(DOMAIN, "speedtest", update)
 
-    hass.async_create_task(
-        async_load_platform(hass, Platform.SENSOR, DOMAIN, {}, config)
+    hass.data.setdefault(DOMAIN, {})
```

Comment:
Before this saved `SpeedtestData(hass)` but no longer so does it actually work?

Reply:
Yes, it's fully operational, even with the split. Is there any evidence I should deliver?

Reply:
No, found it so this line can be just removed

Reply:
I see. Are you sure you're checking the latest commit(s)? Whereas this shouldn't be in (anymore).

![image](https://github.com/home-assistant/core/assets/5011203/287b7301-6a39-48f1-a99d-a5f966fbf266)

Reply:
`hass.data.setdefault(DOMAIN, {})` is still here which is not needed so it can be removed